dbcon/joblist/subquerystep.cpp

```diff
@@ -423,6 +423,9 @@ void SubAdapterStep::execute()
 
                 rowIn.nextRow();
             }
+            // MCOL-3879
+            if (usesFE && rowFe.usesStringTable() && rowFeData.getStringTableMemUsage() > 50 * (1 << 20))
+                rowFeData.clearStringStore();
```

Comment:
We should move this threshold value (50 * (1 << 20)) into a common location (maybe in rowgroup.h) as a const int or a macro, so it can be easily updated later.

Use of UNLIKELY around the if predicate might be a suitable optimization?

We are missing a line break before the if condition.